### PR TITLE
Fixes to maxRows issues + added functionalities

### PR DESCRIPTION
**Pull request by bergben**
With my changes the issues in https://github.com/ManifestWebDesign/angular-gridster/issues/236 are fixed. 
Everything has been tested and proved to work. 
- I created the ability to set **"fixRows"** as an option, which - if set to true - keeps the gridster's height to represent the maxRows limit at all times. 
- The **gridster's width can be set to "match"** which would automatically calculate the width of the gridster based on columns and the witdth of the columns.
- Furthermore I introduced the ability to set the **"gridsterLayout"** option, which - if set to true - creates a layout using "position:absolute" which provides the ability to **style** the gridster's columns and rows. This is done using a directive.
- Also **column header's** can be set, by simply adding an array of Strings **"headers: ['Heading 1', 'Heading 2', 'etc.']"** as an option, 
- There was a bug when adding an item to the gridster where the item wouldn't be able to fit. An error was thrown, but the item was still added to gridster, pushing around other items and still overlapping them at the very end. This has been replaced with an event, **"gridster-item-unplacable"** and an easy method to listen to this event and reacting by removing the item again is described in the Readme.
- The Readme has also been updated for the added gridsterOpts and especially for the gridsterLayout.
-  The minified versions have aswell been updated.

### CODE REVIEW

File: src/angular-gridster.js [hunk 1]
```diff
@@ -38,8 +41,8 @@
 		}
 	})
 
-	.controller('GridsterCtrl', ['gridsterConfig', '$timeout',
-		function(gridsterConfig, $timeout) {
+	.controller('GridsterCtrl', ['gridsterConfig', '$timeout', '$rootScope',
```
Comment (bergben) [inline src/angular-gridster.js]: rootScope is needed to fire the error event "gridster-item-unplacable"

**Comment by danomatic** [comment]
Can you do the following?

fix the misspelling of "unplaceable"
use dashes to separate words in CSS rather than camelCase "gridLayout" -> "grid-layout"

**Comment by bergben** [comment]
Hey, I fixed those things, thanks for pointing them out!

**Comment by samueljun** [comment]
Update on this?

File: src/angular-gridster.js (hunk 1 above)
Comment (wjaspers) [inline src/angular-gridster.js]: why wouldn't you `$emit` an event from the current `$scope`?
if the page has multiple gridsters, each with an unplaceable listener would fire.

**Comment by sourov009** [comment]
@bergben, @danomatic - is there any plan to get this change merged?

**Comment by bergben** [comment]
No plans from my side anymore to get this merged. The project I was working on required some deep changes and additional funcitionalitites (like even a dropzone etc.), which would be sheer overkill for the purpose of angular gridster. So I stopped contributing those changes...

**Pull request closed without merging**